packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -22,30 +24,48 @@ export class RemoteMessageProcessor {
 		this.opSplitter.clearPartialChunks(clientId);
 	}
 
-	public process(remoteMessage: ISequencedDocumentMessage): ISequencedDocumentMessage {
-		let message = copy(remoteMessage);
-		message = this.opDecompressor.processMessage(message).message;
-		unpackRuntimeMessage(message);
-
-		const chunkProcessingResult = this.opSplitter.processRemoteMessage(message);
-		message = chunkProcessingResult.message;
-		if (chunkProcessingResult.state !== "Processed") {
-			// If the message is not chunked or if the splitter is still rebuilding the original message,
-			// there is no need to continue processing
-			return message;
-		}
+	public process(
+		remoteMessage: ISequencedDocumentMessage,
+		processUngroupedMessage?: (message: ISequencedDocumentMessage) => void,
+	): ISequencedDocumentMessage[] {
+		const result: ISequencedDocumentMessage[] = [];
 
-		const decompressionAfterChunking = this.opDecompressor.processMessage(message);
-		message = decompressionAfterChunking.message;
-		if (decompressionAfterChunking.state === "Skipped") {
-			// After chunking, if the original message was not compressed,
-			// there is no need to continue processing
-			return message;
-		}
+		// Ungroup before processing chunks
+		for (let ungroupedMessage of this.opGroupingManager.ungroupOp(copy(remoteMessage))) {
+			processUngroupedMessage?.(ungroupedMessage);
```

Comment:
I'm not sure I follow why summarizer need to observe op before we decompress it?